silq/pulses/pulse_types.py

```diff
@@ -848,6 +847,249 @@ def get_voltage(self, t: Union[float, Sequence]) -> Union[float, np.ndarray]:
+            properties_str += f', full_duration={self.duration}'
+            properties_str += f', offset={self.offset} V'
+        except:
+            pass
+
+        return super()._get_repr(properties_str)
+
+    def get_voltage(self, t: Union[float, Sequence]) -> Union[float, np.ndarray]:
+        """Get voltage(s) at time(s) t.
+
+        Raises:
+            AssertionError: not all ``t`` between `Pulse`.t_start and
+                `Pulse`.t_stop
+        """
+        assert is_between(t, self.t_start, self.t_stop), \
+            f"voltage at {t} s is not in the time range " \
+            f"{self.t_start} s - {self.t_stop} s of pulse {self}"
+
+        assert self.phase_reference == 'relative', f'Phase_reference must be relative, ' \
+                                                   f'since we define the whole waveform ' \
+                                                   f'in the {self.name}.'
+        t = t - self.t_start
+
+        assert self.amplitudes is not None, f'Pulse {self.name} does not have ' \
+                                            f'specified amplitudes.'
+        assert self.frequencies is not None, f'Pulse {self.name} does not have ' \
+                                             f'specified frequencies.'
+        assert self.durations is not None, f'Pulse {self.name} does not have ' \
+                                           f'specified durations.'
+        assert self.phases is not None, f'Pulse {self.name} does not have ' \
+                                        f'specified phases.'
+        assert len(self.amplitudes) == len(self.frequencies) == len(self.durations) == \
+               len(self.phases), f'Pulse {self.name} does not have equal number of amplitudes,' \
+                                 f' frequencies, durations and phases.'
+
+        if isinstance(t, collections.Iterable):
+            waveform = np.zeros(len(t)) + self.offset
+            for idx, (amplitude, frequency, duration, phase) in enumerate(
+                    zip(self.amplitudes, self.frequencies, self.durations, self.phases)):
+                idx_list = [sum(self.durations[:idx]) <= t_id <= sum(self.durations[:idx + 1])
+                            for t_id in t]
+
+                if self.pulse_type == 'sine':
+                    if self.AM_type == 'square':
+                        waveform[idx_list] += amplitude * np.sin(
+                            2 * np.pi * (frequency * t[idx_list] +
+                                         phase / 360))
+                    elif self.AM_type == 'gauss':
+                        t_start = sum(self.durations[:idx])
+                        t_end = sum(self.durations[:idx + 1])
+                        peak_t = (t_start + t_end) / 2
+                        waveform[idx_list] += amplitude * np.exp(
+                            -8 * (t[idx_list] - peak_t) ** 2 / (duration ** 2)) * np.sin(
+                            2 * np.pi * (frequency * t[idx_list] + phase / 360))
```

Comment:
There is no control for the sharpness of the Gaussian, you just have a fixed -8 factor in here.
Is this standard for pulse-shaping?

Reply:
To be honest, I have no idea where -8 factor came from, maybe some typo or something